### PR TITLE
[DDW-298] Add acceptance tests for update mechanism

### PR DESCRIPTION
**Pull request by MarcusHurney**
This PR adds acceptance tests for the NodeUpdateNotification.js component's rendering behavior when a node update is made available. Two test scenarios include one where the node update is postponed and another when the update is accepted.

## Todo:

- [x] Mocks an API call making a node update available.
- [x] Checks that all UI elements in the notification, including the update's version number, have rendered properly.
- [x] Scenario 1: Tests functionality for postponing (declining) an update.
- [x] Scenario 2: Tests functionality for accepting a node update and that the Daedalus electron window has closed.
- [x] All tests after "Scenario 2" fail 

### Basics

- [x] PR is updated to the most recent version of target branch (and there are no conflicts)
- [x] PR has good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [x] CHANGELOG entry has been added and is linked to the correct PR on GitHub
- [x] Automated tests: All acceptance tests are passing (`npm run test`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`npm run dev`)
- [x] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`npm run package` / CI builds)
- [x] There are no *flow* errors or warnings (`npm run flow:test`)
- [x] There are no *lint* errors or warnings (`npm run lint`)
- [x] Text changes are proofread and approved (Jane Wild)
- [x] There are no missing translations (running `npm run manage:translations` produces no changes)
- [x] UI changes look good in all themes (Alexander Rukin)
- [x] Storybook works and no stories are broken (`npm run storybook`)
- [x] In case of npm dependency changes both `package-lock.json` and `yarn.lock` files are updated

### Code Quality
- [x] Important parts of the code are properly documented and commented
- [x] Code is properly typed with flow
- [x] React components are split-up enough to avoid unnecessary re-rendering
- [x] Any code that only works in Electron is neatly separated from components

### Testing
- [x] New feature / change is covered by acceptance tests
- [x] All existing acceptance tests are still up-to-date
- [x] New feature / change is covered by Daedalus Testing scenario
- [x] All existing Daedalus Testing scenarios are still up-to-date

### After Review:
- [ ] Merge PR
- [ ] Delete source branch
- [ ] Move ticket to `done` on the Youtrack board

### CODE REVIEW

File: features/step_definitions/node-update-notification-steps.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+const ACCEPT_BTN = '.NodeUpdateNotification_acceptButton';
+const DENY_BTN = '.NodeUpdateNotification_denyButton';
+
+When(/^I make a node update available$/, async function () {
+  await this.client.executeAsync((nextVersion, done) => {
+    daedalus.api.ada.setNextUpdate(nextVersion)
+      .then(() => daedalus.stores.NodeUpdateStore.refreshNextUpdate())
+      .then(done)
+      .catch((error) => done(error));
+  }, { version: 50 });
```
Comment (nikolaglumac) [inline features/step_definitions/node-update-notification-steps.js]: @MarcusHurney please extract this version value to a constant which we can use in later steps where you check if the version shown in the notification matches this expected one.

**Comment by nikolaglumac** [comment]
@MarcusHurney looks like all tests after the one which applies the update fail (due to chrome unavailable). We need to fix that!

**Comment by MarcusHurney** [comment]
@nikolaglumac @DominikGuzei 
When you're reviewing this and run **npm run test**, you'll notice that the accept node update scenario shuts down the window and there is a message in the console saying that a session id must be present for one of electron's or spectron's window methods that runs when the window is opening or closing via `app.start()` or maybe `app.quit()`, I'm not sure when it runs. It doesn't cause any of the tests to fail nor does it have any direct consequence on our suite of tests, but just keep any eye out for it and see what you think.

**Comment by DominikGuzei** [comment]
@MarcusHurney is this ready to be reviewed?

**Comment by nikolaglumac** [comment]
@DominikGuzei AFAIK it should be ready. Please see last comment @MarcusHurney posted in this thread...

File: features/support/electron.js [hunk 2]
```diff
@@ -85,6 +85,13 @@ defineSupportCode(({ BeforeAll, Before, After, AfterAll, setDefaultTimeout }) =>
     });
   });
 
+  // this ensures that the spectron instance of the app restarts
+  // after the node update acceptance test shuts it down via 'kill-process'
+  // eslint-disable-next-line prefer-arrow-callback
+  After({ tags: '@preventAppQuit' }, async function () {
```
Comment (DominikGuzei) [inline features/support/electron.js]: maybe `@preventAppQuit` should be renamed to `@restartApp`

File: source/main/ipc-api/kill-process.js [hunk 3]
```diff
@@ -1,10 +1,10 @@
 // @flow
-import { ipcMain } from 'electron';
+import { ipcMain, app } from 'electron';
 
 export default () => {
   ipcMain.on('kill-process', () => {
+    app.quit();
     // if (event.sender !== mainWindow.webContents) return;
-    // TODO: fix this
-    process.exit(20);
```
Comment (DominikGuzei) [inline source/main/ipc-api/kill-process.js]: @MarcusHurney please be careful with this change - exit code 20 might be necessary for cardano-launcher … @cleverca22 can you take a look at this?

Comment (MarcusHurney) [reply source/main/ipc-api/kill-process.js]: @DominikGuzei 
Changing this line to **app.quit** ended up being unnecessary so I've changed it back. All tests still pass.

File: features/support/electron.js (hunk 2 above)
Comment (MarcusHurney) [reply features/support/electron.js]: Your request has been processed!

**Comment by DominikGuzei** [comment]
@nikolaglumac from my side this PR is ready for merging!

**Pull request merged**